pootle/apps/pootle_project/models.py

```diff
@@ -420,6 +423,13 @@ def delete(self, *args, **kwargs):
 
         directory.delete()
 
+    def clean(self):
+        if not self.code:
+            raise ValidationError('A code must be provided.')
```

Comment:
Without this, one can create projects without code. Such a project would be useless and accidental, but it's technically possible.